### PR TITLE
fix: Set the limit of Occurrences

### LINKED ISSUE #10204: App is crashing when a recurring event type is set with negative occurrences
### Issue Summary

The recurring event types are accepting negative values as the number of occurrences. This situation causes the app to crash. Same if you type a big number.

### Steps to Reproduce

1. Create an Event-type and set a Recurring for that
2. Open a preview for that event
3. Write `-1` or use the arrow down from the component to change to a negative number
4. Select an available date
5. Select an available time
6. Observe

### Actual Results

  - Recurring event type is accepting dash and plus sign. It allows the user to set a negative value and it causes a crash

### Expected Results

  - The occurrences field should only accept numbers and cannot be more than the maximum set in the configuration.

### Technical details

  - Tested on `app.cal.dev`
  - Tested on: Firefox (v115.0.2) and Chrome (v114.0.5735.199)
 
### Evidence

image

### LINKED ISSUE #10207: Recurring event-type booking is not responsive
### Issue Summary

When booking a recurring event-type, if the page dimensions change and back to normal, the recurring set is back to the maximum occurrences set.

### Steps to Reproduce

1. Go to book a recurring event with, for example, 12 occurrences, using a normal dimension
2. Set a number of occurrences, like 2
3. Decrease the page dimensions till the occurrence labels dismiss
4. Increase the page dimensions till the occurrence labels appear

### Actual Results

  - The number of occurrences is set back to the maximum occurrences set

### Expected Results

  - The occurrence number should be kept

### Technical details

  - Tested on `app.cal.dev`
  - Tested on: Firefox (v115.0.2) and Chrome (v114.0.5735.199)

### Evidence

https://www.loom.com/share/ac7d10c819ff4fceaf0f4b8e421896ce

### LINKED ISSUE #10280: Occurrence value is changed to maximum when came back
Found a bug? Please fill out the sections below. 👍

### Issue Summary

After changing the occurrences when back to the booking page to change the date or time, occurrence value is changed to maximum

Loom Video:- https://www.loom.com/share/6eafea1ea8794356b7575e733514a5c4?sid=76295ccc-bd3b-4f3a-8764-f0186fe39d30
### Steps to Reproduce

1. Go to book a recurring event with, for example, 1 occurrence.
2. Set a number of occurrences, like 2
3. select date and time
4. came back to booking page

Any other relevant information. For example, why do you consider this a bug and what did you expect to happen instead?

### Actual Results
occurrence value set to maximum.
### Expected Results

The value should be the input value or which is set by user.

### PR DESCRIPTION
**Pull request by anikdhabal**
## What does this PR do?

A warning alert will be visible if the input value is not a number and between the range. If the value is not between the range, then the occurrence value is set to default value. And also it is responsive.

Fixes #10204 
Fixes #10280
Fixes #10207 

loom video:- https://www.loom.com/share/bf6253b371f447409ce6c5ffdcf1ad48?sid=964b7646-bf0e-43aa-9a54-5e449d58317f

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)

### CODE REVIEW

File: apps/web/public/static/locales/en/common.json [hunk 1]
```diff
@@ -507,6 +507,7 @@
   "your_name": "Your name",
   "your_full_name": "Your full name",
   "no_name":"No name",
+  "enter_number_between_range": "Please enter a number between 1 and 20.",
```
Comment (emrysal) [inline apps/web/public/static/locales/en/common.json:510]: I think this should be a variable based on the maximum occurences set by the event type?

Comment (anikdhabal) [reply apps/web/public/static/locales/en/common.json:510]: okk

Comment (anikdhabal) [reply apps/web/public/static/locales/en/common.json]: Done

Comment (anikdhabal) [reply apps/web/public/static/locales/en/common.json]: > I think this should be a variable based on the maximum occurences set by the event type?

Done

**Review by CarinaWolli — APPROVED** [review]
Looks good! Thank you 🙏

**Pull request merged**